jdaviz/app.py

```diff
@@ -861,26 +863,36 @@ def get_subsets(self, subset_name=None, spectral_only=False,
             if isinstance(subset.subset_state, CompositeSubsetState):
                 # Region composed of multiple ROI or Range subset
                 # objects that must be traversed
-                subset_region = self.get_sub_regions(subset.subset_state)
+                subset_region = self.get_sub_regions(subset.subset_state, simplify_spectral)
             elif isinstance(subset.subset_state, RoiSubsetState):
                 # 3D regions represented as a dict including an
                 # AstropyRegion object if possible
                 subset_region = self._get_roi_subset_definition(subset.subset_state)
             elif isinstance(subset.subset_state, RangeSubsetState):
                 # 2D regions represented as SpectralRegion objects
-                subset_region = self._get_range_subset_bounds(subset.subset_state)
+                subset_region = self._get_range_subset_bounds(subset.subset_state,
+                                                              simplify_spectral)
             else:
                 # subset.subset_state can be an instance of MaskSubsetState
                 # or something else we do not know how to handle
                 all_subsets[label] = None
                 continue
 
-            if isinstance(subset_region, SpectralRegion):
+            # Is the subset spectral or spatial?
+            is_spectral = self._is_subset_spectral(subset_region)
+
+            # Remove duplicate spectral regions
+            if is_spectral and isinstance(subset_region, SpectralRegion):
                 subset_region = self._remove_duplicate_bounds(subset_region)
+            elif is_spectral:
+                subset_region = self._remove_duplicate_bounds_in_dict(subset_region)
 
-            if spectral_only and isinstance(subset_region, SpectralRegion):
-                all_subsets[label] = subset_region
-            elif spatial_only and not isinstance(subset_region, SpectralRegion):
+            if spectral_only and is_spectral:
+                if object_only and not simplify_spectral:
+                    all_subsets[label] = [reg['region'] for reg in subset_region]
+                else:
+                    all_subsets[label] = subset_region
+            elif spatial_only and not is_spectral:
```

Comment:
Why would something be spatial and spectral at the same time? Pondering for another day.

Reply:
`is_spectral` is a bool for whether the subset in question is spectral or not. So if the `spatial_only` flag is true and `is_spectral` is False then we add the subset information to the returned list.